File: Haskell/day1.hs
import System.IO

combinations :: Int -> [a] -> [[a]] 
combinations 0 _ = [[]]
combinations _ [] = [[]]
combinations n (x:xs) = map (x:) (combinations (n-1) xs) ++ combinations n xs

tally :: Int -> [Int] -> Int
tally _ [] = 0
tally n xs = head [product x | x <- combinations n xs, sum x == 2020]

main :: IO ()
main = do
    content <- readFile "Day1Input.txt"
    let nlist = map read $ words content :: [Int]
    print $ tally 2 nlist
    print $ tally 3 nlist

File: Python/day1.py
#!/usr/bin/env python
from itertools import combinations
from math import prod

def tally(nlist, n):
    return next(prod(i) for i in combinations(nlist, n) if sum(i) == 2020)

def main():
    with open(f'{__file__[:-3].capitalize()}Input.txt', 'r') as ifile:
        nlist = list(map(int, ifile.readlines()))

    print(tally(nlist, 2)) # 1
    print(tally(nlist, 3)) # 2
